crates/schema/src/auto_migrate.rs

```diff
@@ -604,24 +604,19 @@ fn auto_migrate_view<'def>(plan: &mut AutoMigratePlan<'def>, old: &'def ViewDef,
     })
     .collect();
 
-    // TODO: Uncomment and re-enable view auto-migrations without disconnecting clients
-    //
-    // if old.is_anonymous != new.is_anonymous || incompatible_return_type || incompatible_param_types {
-    //     plan.steps.push(AutoMigrateStep::AddView(new.key()));
-    //     plan.steps.push(AutoMigrateStep::RemoveView(old.key()));
-
-    //     if !plan.disconnects_all_users() {
-    //         plan.steps.push(AutoMigrateStep::DisconnectAllUsers);
-    //     }
-    // } else {
-    //     plan.steps.push(AutoMigrateStep::UpdateView(old.key()));
-    // }
-
-    plan.steps.push(AutoMigrateStep::AddView(new.key()));
-    plan.steps.push(AutoMigrateStep::RemoveView(old.key()));
+    if old.is_anonymous != new.is_anonymous
+        || incompatible_return_type
+        || incompatible_param_types
+        || old.return_type != new.return_type
+    {
+        plan.steps.push(AutoMigrateStep::AddView(new.key()));
+        plan.steps.push(AutoMigrateStep::RemoveView(old.key()));
 
-    if !plan.disconnects_all_users() {
-        plan.steps.push(AutoMigrateStep::DisconnectAllUsers);
+        if !plan.disconnects_all_users() {
+            plan.steps.push(AutoMigrateStep::DisconnectAllUsers);
+        }
+    } else {
+        plan.steps.push(AutoMigrateStep::UpdateView(old.key()));
```

Comment:
```suggestion
        plan.steps.push(AutoMigrateStep::UpdateView(new.key()));
```
I think this was me. Probably should use the new key.